mantidproject/mantid
# Check Data Directories on Request, Not Startup

**Description of work.**

Changes the behaviour of the ConfigService so that data directories are checked when access is reqested, rather than on startup. The intention is to reduce the startup time of workbench when checking for network drives that may be very slow to access. 

**To test:**
1. Launch workbench
2. Add some directories on network drives to the list of user data directories.
3. Slow down the network connection somehow (utilities such as `trickle` might be useful)
4. Relaunch workbench, it should start more quickly as it isn't waiting for connections on startup any more. 

Refs #28499 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
